docs/enhancements/fep-001-api-fleet-templating.md

```diff
@@ -0,0 +1,257 @@
+  the fleet template re-rendered. (this is the same workflow as when the labels selected by a fleet changes)
+
+### User Stories 
+
+#### Story 1: network access credentials
+As an IT administrator, I want to use site-specific network access credentials for devices in the same fleet, so that I do not have to deal with the complexity of maintaining one fleet per site.
+
+#### Story 2: site specific endpoints
+As an IT administrator, I want to provide site-specific server endpoints to devices that depend on LAN-configured edge servers.
+
+#### Story 3: site specific certificate bundles
+As an IT administrator, I want to provide site-specific certificate bundles to enable trust with the local site infrastructure.
+
+#### Story 4: injecting the device name into the configuration
+As an IT administrator, I want the name of my MicroShift instance in ACM to have the same cluster name as the underlying device's name or fingerprint, so I can easily correlate between the two.
+
+#### Story 5: site specific telemetry endpoints
+As an IT administrator, I want to provide site-specific/region-specific telemetry endpoints to devices.
+
+#### Story 6: dialing up verbosity for logging in my user application in specific devices
+As an IT administrator, I want to dial up the verbosity for logging on a specific device, so I can troubleshoot it.
+
+#### Story 7: region specific application service endpoints
+As an IT administrator, I want to provide region-specific service endpoints to my applications
+
+#### Story 8: region specific radio bands/regulations
+As a developer, I want to ensure that my devices automatically comply with local regulations when enrolled in specific regions. (i.e. configured with region-specific radio bands for WiFi, 5G, or other radio technologies)
+
+#### Story 9: region specific data sovereignty
+As a developer, I want to ensure that my devices connect to the specific regional endpoints to ensure compliance with data sovereignty.
+
+### Notes/Constraints/Caveats
+
+We would need to identify the design changes relative to templateVersions and
+if the relationship of those continues to be to the fleet or device level.
+
+### Risks and Mitigations
+
+
+## Design Details
+
+TBD
+
+### Scalability
+
+TBD
+
+### Troubleshooting
+
+Troubleshooting can be done by looking at the device status and rendered device spec. Any errors found in templates
+could be reported as part of the fleet status conditions.
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+### Every configuration combination has it's own fleet
+This is the most obvious alternative, this proposal is not implemented, and administrators
+take care of replicating fleet definitions as necessary to accomodate the different
+configurations.
+
+This approach does not allow customization based on fields like the device name.
+
+### Fleet template inheritance / sub-fleets
```

Comment:
If you have a fleet of cars with various manufacture dates/capabilities. Would this require a fleet for every car model/year? It seems that templating solves some but not all of these problems.  An intermediary abstraction might also be required. But I would agree in the happy path templating solves problems sub-fleets do not.

Reply:
Yes, it would require a separate fleet for each end variant in the end.

The main proposal is better in this aspect.

I will capture this.

Reply:
OK so then as a user  `"how can I efficiently manage many fleets"` is maybe a good question we can flesh out.

Reply:
Fleets of fleets ;)